tests/unittests/tests-gcoap/tests-gcoap.c

```diff
@@ -151,6 +151,27 @@ static void test_gcoap__client_put_req(void)
     TEST_ASSERT_EQUAL_INT('1', (char)*pdu.payload);
 }
 
+/*
+ * Builds on client_put_req to test overfill on coap_opt_finish().
+ */
+static void test_gcoap__client_put_req_overfill(void)
+{
+    /* header 4, token 2, path 11, format 1, marker 1 = 19*/
+    uint8_t buf[18+GCOAP_REQ_OPTIONS_BUF];
+    coap_pkt_t pdu;
+    ssize_t len;
+    char path[] = "/riot/value";
+
+    gcoap_req_init(&pdu, buf, 18+GCOAP_REQ_OPTIONS_BUF, COAP_METHOD_PUT, path);
```

Comment:
use `sizeof(buf)` here, instead of `18+GCOAP...`?

Reply:
Yep, definitely a better approach.